homeassistant/components/isy994/__init__.py

```diff
@@ -139,31 +146,50 @@ async def async_setup_entry(
     if host.scheme == "http":
         https = False
         port = host.port or 80
+        session = aiohttp_client.async_create_clientsession(
+            hass, verify_ssl=None, cookie_jar=CookieJar(unsafe=True)
+        )
     elif host.scheme == "https":
         https = True
         port = host.port or 443
+        session = aiohttp_client.async_get_clientsession(hass)
     else:
         _LOGGER.error("The isy994 host value in configuration is invalid")
         return False
 
     # Connect to ISY controller.
-    isy = await hass.async_add_executor_job(
-        partial(
-            ISY,
-            host.hostname,
-            port,
-            username=user,
-            password=password,
-            use_https=https,
-            tls_ver=tls_version,
-            webroot=host.path,
-        )
+    isy = ISY(
+        host.hostname,
+        port,
+        username=user,
+        password=password,
+        use_https=https,
+        tls_ver=tls_version,
+        webroot=host.path,
+        websession=session,
+        use_websocket=True,
     )
-    if not isy.connected:
-        return False
 
-    # Trigger a status update for all nodes, not done automatically in PyISY v2.x
-    await hass.async_add_executor_job(isy.nodes.update)
+    try:
+        with async_timeout.timeout(30):
+            await isy.initialize()
+    except ISYInvalidAuthError as err:
+        _LOGGER.error(
+            "Invalid credentials for the ISY, please adjust settings and try again: %s",
+            err,
+        )
+        return False
```

Comment:
We can do raise ConfigEntryAuthFailed in the future

Reply:
Noted.